doc/classes/bool.xml

```diff
@@ -4,7 +4,55 @@
 		Boolean built-in type.
 	</brief_description>
 	<description>
-		Boolean built-in type.
+		Boolean is built-in type. It can represent any data type that is either a true or false value. You can think of it as on or off (1 or 0) switch. It's often used as part of programming logic in condition statements like `if` statements.
+
+		For example:
+
+		[codeblock]
+		var can_shoot = true # Assigns true value to can_shoot.
+
+
+		func shoot():
+        	if can_shoot:
+            	# Perform shooting actions here.
+		[/codeblock]
+
+		As you can see above `if can_shoot` is equivalent of `if can_shoot == true`. It is good practice to follow the natural spoken language structure when possible. Use `if can_shoot` rather than `if can_shoot == true` and use `if not can_shoot` rather than `if can_shoot == false`.
+
+		You can use booleans to create more complex logic for example:
+
+		[codeblock]
+		var can_shoot = true # Assigns true values to can_shoot
+
+
+		func shoot():
+			if can_shoot and Input.is_action_pressed("shoot"):
+				create_bullet()
+		[/codeblock]
+
+		Please note that `Input.is_action_pressed("shoot")` is also a boolean that is true when "shoot" is pressed and false when `shoot` isn't pressed.
+
+		The above code will only create a bullet if both condition are met action "shoot" is pressed and if can_shoot is true. One of practical use cases could be implementing a cool down on player weapon.
+
+		For example above code could be expanded further to:
+
```

Comment:
Remove this line.

Reply:
Don't you think it's a good example for a reader that booleans are used more often and don't have to appear in form of `my_bool = true`?

Reply:
I don't understand your question.

Reply:
You said remove this line. I don't know why or am I getting confused and this comment is no longer up to date?

Reply:
The lines that I'm referring to be removed are the empty ones.